python/hail/expr/types.py

```diff
@@ -627,6 +628,75 @@ def __str__(self):
         return "TStruct({}, {})".format(repr(list(names)), repr(list(types)))
 
 
+class TTuple(Type):
+    """Hail type for tuples.
+
+    In Python, these are represented as :obj:`tuple`.
+
+    Parameters
+    ----------
+    types: varargs of :class:`.Type`
+        Element types.
+    """
+
+    @typecheck_method(types=Type)
+    def __init__(self, *types):
+        self._get_jtype = lambda: scala_object(Env.hail().expr.types, 'TTuple').apply(map(lambda t: t._jtype, types),
+                                                                                      False)
+        self._types = types
+        super(TTuple, self).__init__()
+
+    @property
+    def types(self):
+        """Tuple element types.
+
+        Returns
+        -------
+        :obj:`list` of :class:`.Type`
```

Comment:
this is a tuple (it comes from the varargs). That's fine, though (it means users can't append to it)